src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -211,19 +207,53 @@ void OperationalStateServer::HandleCommand(HandlerContext & handlerContext, Func
     }
 }
 
-CHIP_ERROR OperationalStateServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+// This function is called by the interaction model engine when a command destined for this instance is received.
+void Instance::InvokeCommand(HandlerContext & handlerContext)
 {
+    ChipLogDetail(Zcl, "OperationalState: InvokeCommand");
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::Pause::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Pause state");
+
+        HandleCommand<Commands::Pause::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandlePauseState(ctx, req); });
+        break;
+
+    case Commands::Resume::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Resume state");
+
+        HandleCommand<Commands::Resume::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleResumeState(ctx, req); });
+        break;
+
+    case Commands::Start::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Start state");
+
+        HandleCommand<Commands::Start::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleStartState(ctx, req); });
+        break;
+
+    case Commands::Stop::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Stop state");
+
+        HandleCommand<Commands::Stop::DecodableType>(handlerContext,
+                                                     [this](HandlerContext & ctx, const auto & req) { HandleStopState(ctx, req); });
+        break;
+    }
+}
+
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    ChipLogError(Zcl, "OperationalState: Reading");
     switch (aPath.mAttributeId)
     {
     case OperationalState::Attributes::OperationalStateList::Id: {
-        Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
-        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is nullptr"));
-
-        return aEncoder.EncodeList([delegate](const auto & encoder) -> CHIP_ERROR {
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
```

Comment:
Or could do:
```suggestion
        return aEncoder.EncodeList([delegate = mDelegate](const auto & encoder) -> CHIP_ERROR {
```
but either way.

Reply:
Accepted.